libs/state/src/lib/rx-state.service.ts

```diff
@@ -547,7 +587,13 @@ export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
     obsOrObsWithSideEffect: Observable<S>,
     sideEffectFn?: (arg: S) => void
   ): void {
-    const sideEffect = obsOrObsWithSideEffect.pipe(catchError(e => EMPTY))
+    const sideEffect = obsOrObsWithSideEffect.pipe(
+      catchError((e) => {
+        const errorHandler = inject(ErrorHandler, InjectFlags.Optional);
```

Comment:
Yeah, it actually will always fail here. Providing `InjectFlags` is a "hack" to make the runtime behaviour happy (since the `inject must be called blah-blah...` error is not thrown). I also can't think of a situation where `ErrorHandler` can be `null` since it's initially provided by Angular or overwritten by developers.

Reply:
If injecting the ErrorHandler using the constructor:
```
constructor(private errorHandler: ErrorHandler) {}
```
will not be backward compatible with -  https://github.com/rx-angular/rx-angular/blob/master/libs/state/docs/usage.md#inherit
```
@Directive({
  selector: '[appStateful]',
})
export class StatefulComponent extends RxState<{ foo: number }> {
  readonly state$ = this.select();

  constructor(private errorHandler: ErrorHandler) {
    super(this.errorHandler);
  }
}
```
how can this be fixed?

Reply:
That's a use case I didn't think about, then I would say go for the injector usage and document why in a comment.